### PR TITLE
feat(ui-ux): Automatic display of warning message (No blocks minted > 45 mins) 

### LINKED ISSUE #856: Automatic display of warning message (No blocks minted > 45 mins)
#### What would you like to be added:
Similar to https://github.com/DeFiCh/wallet/issues/1849. An automatic message should be displayed when there is no new blocks minted for **45 mins**.

**Display Message**

```
We are currently investigating a syncing issue on the blockchain. View more details on the [DeFiChain Status Page](https://status.defichain.com/).
```

Note:
- If possible try explore ways to reuse data that is already stored in the state to reduce number of api calls. It may be possible to use data from stats. However, it may not contain the last block datetime.

/triage accepted
/area ui-ux

### PR DESCRIPTION
**Pull request by chloezxyy**
#### What this PR does / why we need it:
An automatic message should be displayed when there is no new blocks minted for 45 mins.

#### Which issue(s) does this PR fixes?:

Fixes #856

#### Sample Links & Screenshots:

Link: 
<details>
<summary>Desktop Screenshot</summary>

image
image

</details>

<details>
<summary>Mobile Screenshot</summary>

</details>

#### Developer Checklist:

- [x] Read your code changes at least once
- [ ] Tested on multiple web browsers
- [x] Tested responsiveness (e.g, iPhone, iPad, Desktop)
- [x] No console errors
- [ ] Unit tests*
- [x] Added e2e tests*

### CODE REVIEW

File: src/components/index/IndexHeader.tsx [hunk 1]
```diff
@@ -71,15 +71,15 @@ function Stats (): JSX.Element {
         <ReactNumberFormat
           displayType='text'
           thousandSeparator
-          value={stats.price.usd}
+          value={stats?.price?.usd}
```
Comment (pierregee) [inline src/components/index/IndexHeader.tsx:74]: hmm, this is not throwing an error on my end and checking the types, it should be fine without the optional chaining

**Review by pierregee — CHANGES_REQUESTED** [review]

File: cypress/e2e/components/WarningBanner.spec.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+      statusCode: 200,
+      data: {
+        lastSync: new Date().toString(),
+        lastSuccessfulSync: new Date().toString()
+      }
+    })
+    cy.findByTestId('warning_banner').should('not.exist')
+  })
+
+  it.only('should display blockchain is down warning banner after preset interval', () => {
```
Comment (pierregee) [inline cypress/e2e/components/WarningBanner.spec.ts:37]:
```suggestion
  it('should display blockchain is down warning banner after preset interval', () => {
```

File: src/components/commons/banner/WarningBanner.tsx [hunk 3]
```diff
@@ -8,24 +8,59 @@ interface WarningBannerProps {
...
+}
+
 export function WarningBanner (props: PropsWithChildren<WarningBannerProps>): JSX.Element {
-  const network = useNetwork().name
+  const deFiChainStatusUrl = 'https://status.defichain.com/'
 
-  if (props.mainnet === undefined && props.testnet === undefined) {
-    return <></>
+  const blockchainIsDownContent: Announcement = {
+    content: 'We are currently investigating a syncing issue on the blockchain. ',
```
Comment (thedoublejay) [inline src/components/commons/banner/WarningBanner.tsx:20]:
```suggestion
    content: 'We are currently investigating a syncing issue on the blockchain.',
```

File: src/components/commons/banner/WarningBanner.tsx [hunk 4]
```diff
@@ -8,24 +8,59 @@ interface WarningBannerProps {
...
+    url: deFiChainStatusUrl
   }
-
-  if (network !== 'mainnet' && props.mainnet!) {
-    return <></>
+  const oceanIsDownContent: Announcement = {
+    content: 'We are currently investigating connection issues on Ocean API. ',
```
Comment (thedoublejay) [inline src/components/commons/banner/WarningBanner.tsx:24]: why is there a space at the end haha

File: src/components/commons/banner/WarningBanner.tsx [hunk 5]
```diff
@@ -8,24 +8,59 @@ interface WarningBannerProps {
...
+
+  // don't display banner when there is not announcement to display
+  if (displayAnnouncement === undefined) {
     return <></>
   }
 
   return (
-    <div className='bg-orange-100 rounded p-3 text-center' data-testid={props.testId}>
-      {props.children}
+    <div className='bg-orange-100 rounded p-3 text-center flex-row text-center' data-testid='warning_banner'>
```
Comment (thedoublejay) [inline src/components/commons/banner/WarningBanner.tsx:47]: Can we replace this to HTML5 elements instead?

File: src/components/commons/banner/WarningBanner.tsx [hunk 6]
```diff
@@ -8,24 +8,59 @@ interface WarningBannerProps {
...
+      <div>
+        {displayAnnouncement.content}
```
Comment (thedoublejay) [inline src/components/commons/banner/WarningBanner.tsx:49]: Also, you can put this outside, so you don't need to add space on label. Just add padding/margin

**Review by thedoublejay — CHANGES_REQUESTED** [review]

File: src/components/index/IndexHeader.tsx (hunk 1 above)
Comment (chloezxyy) [reply src/components/index/IndexHeader.tsx:74]: As discussed, optional chaining will be required as a safety net against undefined data retrieved from the api

File: src/layouts/components/HeaderCountBar.tsx [hunk 7]
```diff
@@ -73,22 +73,22 @@ export function HeaderCountBar (props: { className: string }): JSX.Element {
       <div
         className='px-4 py-3 font-normal text-sm bg-white text-left text-gray-900 rounded-lg border border-gray-100 shadow-md '
       >
-        <PopoverTVLAmount text='DEX' count={tvl.dex} />
-        <PopoverTVLAmount text='Masternode' count={tvl.masternodes} />
-        <PopoverTVLAmount text='Vault Collateral' count={tvl.loan} />
-        <PopoverTVLAmount text='Total' count={tvl.total} />
+        <PopoverTVLAmount text='DEX' count={tvl?.dex} />
```
[the remark this answers is not in the document]
Comment (chloezxyy) [reply src/layouts/components/HeaderCountBar.tsx:76]: As discussed, other than for e2e, optional chaining will be required as a safety net against undefined data retrieved from the api

**Review by pierregee — APPROVED** [review]
@thedoublejay @chloezxyy  I'm good with the PR!
There's one scenario in e2e that isn't covered tho because we don't have a setup for e2e with SSR

**Covered Scenario: /stats is down**
Current flow of e2e
1. SSR renders the page (gets `/stats` from playground) - success
2. e2e intercepts `/stats` on CSR - gets the intercept response from e2e
3. The banner displays

^ This is related to the additional optional chaining added on the code.

**Uncovered scenario: /stats is down****
1. SSR renders the page - /stats is down (cannot intercept because it's in SS)
2. e2e intercepts `/stats` on CSR - gets the intercept response from e2e
3. The banner displays

**Review by thedoublejay — APPROVED** [review]

**Pull request merged**